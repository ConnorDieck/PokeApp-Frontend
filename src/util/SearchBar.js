import { Select, TextField, MenuItem, Button, FormLabel } from "@mui/material";
import React, { useState } from "react";

const SearchBar = ({ search }) => {
	let INITIAL_STATE = {
		name : "",
		type : null
	};

	const [ fData, setFormData ] = useState(INITIAL_STATE);

	const types = [
		"Normal",
		"Fire",
		"Water",
		"Grass",
		"Electric",
		"Ice",
		"Fighting",
		"Poison",
		"Ground",
		"Flying",
		"Psychic",
		"Bug",
		"Rock",
		"Ghost",
		"Dark",
		"Dragon",
		"Steel",
		"Fairy"
	];

	const handleChange = evt => {
		const { name, value } = evt.target;
		setFormData(fData => ({
			...fData,
			[name] : value
		}));
	};

	// When submitted, runs search function passed down from SpeciesList to set species to the result in state
	const handleSubmit = evt => {
		evt.preventDefault();
		search(fData);
		setFormData(INITIAL_STATE);
	};

	return (
		<form className="searchBar" onSubmit={handleSubmit}>
			<TextField //
				id="name"
				name="name"
				value={fData.name}
				onChange={handleChange}
				placeholder="Name..."
			/>
			<FormLabel htmlFor="type">Type:</FormLabel>
			<Select //
				id="type"
				name="type"
				value={fData.type}
				onChange={handleChange}
				placeholder=""
			>
				{types.map(t => <MenuItem value={t}>{t}</MenuItem>)}
			</Select>
			<Button color="secondary" variant="contained" type="submit">
				Search
			</Button>
		</form>
	);
};

export default SearchBar;
