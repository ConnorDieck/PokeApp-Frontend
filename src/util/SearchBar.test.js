import React from "react";
import { render } from "@testing-library/react";
import SearchBar from "./SearchBar";
import { MemoryRouter } from "react-router-dom";

//smoke test
it("renders without crashing", function() {
	render(
		<MemoryRouter>
			<SearchBar />
		</MemoryRouter>
	);
});

// snapshot test
test("matches snapshot", function() {
	const { asFragment } = render(
		<MemoryRouter>
			<SearchBar />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
