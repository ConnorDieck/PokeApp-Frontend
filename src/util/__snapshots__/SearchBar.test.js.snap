// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="searchBar"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="name"
          name="name"
          placeholder="Name..."
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <label
      class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
      for="type"
    >
      Type:
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="type"
        class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="type"
        role="button"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        name="type"
        placeholder=""
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-hdw1oc"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Search
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
